src/native-state-wayland.cpp

```diff
@@ -121,6 +143,31 @@ NativeStateWayland::registry_handle_global(void *data, struct wl_registry *regis
 
         wl_output_add_listener(my_output->output, &output_listener_, my_output);
         wl_display_roundtrip(that->display_->display);
+    } else if (strcmp(interface, "wl_seat") == 0) {
+        that->display_->seat =
+            static_cast<struct wl_seat *>(
+            wl_registry_bind(registry, id, &wl_seat_interface, 1));
+        wl_seat_add_listener(that->display_->seat, &seat_listener_, that);
+    } else if (strcmp(interface, "wl_shm") == 0) {
+        that->display_->shm =
+            static_cast<struct wl_shm *>(
+                wl_registry_bind(registry, id, &wl_shm_interface, 1));
+
+        struct my_cursor *my_cursor = new struct my_cursor();
+        my_cursor->cursor_surface =
+             wl_compositor_create_surface(that->display_->compositor);
+        my_cursor->cursor_theme = wl_cursor_theme_load(NULL, 32, that->display_->shm);
+
+        if (!my_cursor->cursor_theme) {
+            fprintf(stderr, "unable to load default theme\n");
```

Comment:
We can use the the Log module to print out errors (i.e., `Log::error(...)`)

Reply:
Added `Log::error()` instead of `fprintf()`